File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int i = -1, j = -1, k = 0;
        for (; k < nums.size(); ++k) {
            if (i == -1 or nums[k] <= nums[i]) {
                i = k;
            } else if (j == -1 or nums[k] <= nums[j]) {
                j = k;
            } else {
                return true;
            }
        }
        return false;
    }
};

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        i: int = -1
        j: int = -1
        for k in range(len(nums)):
            if (i == -1 or nums[k] <= nums[i]):
                i = k
            elif (j == -1 or nums[k] <= nums[j]):
                j = k
            else:
                return True
        return False